Azure/sonic-buildimage
# [Build] Improve docker build performance

#### Why I did it
The docker storage driver vfs is not a good option for build, it uses the “deep copy” when building a new layer, leads to lower performance and more space used on disk than other storage drivers.
A better docker storage driver is the default one overlay2, it is a modern union filesystem.

1. Improve build performance, reduce the build time
    Reduce around 3 hours in Broadcom official build, from 12.5 hours to 9.5 hours.
2. Reduce the disk write bytes
    Reduce from 1600G to 360G in Broadcom official build.
3. Reduce the required disk size during the build.

#### How I did it
1. Change the storage driver setting of the docker in slave container.
2. Mount the local file system to /var/lib/docker, since the overlay fs as the backing filesystem is not supported.



#### How to verify it

Verify the build option:
For default option:
BLDENV=buster make -f Makefile.work docker-start KEEP_SLAVE_ON=yes
```
xumia@dccae0cd559c:/sonic$ docker info | grep "Storage Driver"
Storage Driver: vfs
```
Enable overlay2:
SONIC_SLAVE_DOCKER_DRIVER=overlay2 BLDENV=buster make -f Makefile.work docker-start KEEP_SLAVE_ON=yes
```
xumia@ba4b89040dbb:/sonic$ docker info | grep "Storage Driver"
Storage Driver: overlay2
```
Compare the disk write bytes metrics:
<img width="873" alt="image" src="https://user-images.githubusercontent.com/59720581/173164861-1adb30a3-1fb3-410b-aa26-87a146af57c8.png">


#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog


#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
